src/scripts/branch_review.py

```diff
@@ -0,0 +1,312 @@
+        obsoletion_candidates_children_relationships = list(
+            OI.relationships(
+                objects=obsoletion_candidates_set, predicates=[IS_A, PART_OF]
+            )
+        )
+
+        # For each branch_id
+        for branch_id in branch_ids:
+            # Get the descendants of the current branch_id
+            branch_descendants_set = branch_descendants_dict[branch_id]
+            obsoletion_candidate_child_of_branch = (
+                branch_descendants_set & obsoletion_candidates_set
+            )
+
+            children_of_obsoletion_candidate = set()
+
+            for obsoletion_candidate in obsoletion_candidate_child_of_branch:
+                # Column E: Parent to be obsoleted == obsoletion_candidate
+                children_of_obsoletion_candidate.update(
+                    set(
+                        get_immediate_children(
+                            curie=obsoletion_candidate,
+                            relationships=obsoletion_candidates_children_relationships,
+                        )
+                    )
+                )
+
+        children_of_obsoletion_candidate_parents_relationship = set(
+            OI.relationships(
+                subjects=children_of_obsoletion_candidate,
+                predicates=[IS_A, PART_OF],
+            )
+        )
+
+        children_also_obsoletion_candidates = (
+            children_of_obsoletion_candidate & obsoletion_candidate_child_of_branch
+        )
+
+        for child in children_of_obsoletion_candidate:
+            # Column B: Children of ObsoleteCandidate == child
+            parents_of_child = (
+                set(
+                    get_immediate_parent(
+                        curie=child,
+                        relationships=children_of_obsoletion_candidate_parents_relationship,
+                    )
+                )
+                - all_obsoletes
+            )
+
+            # These are parents that need to be obsoleted and hence NOT in Branch
+            relevant_obsoletion_parents_in_branch = (
+                parents_of_child & obsoletion_candidate_child_of_branch
+            )
+            # if child == "MONDO:0013310":
+            #     import pdb; pdb.set_trace()
+
+            # parents_of_child.difference_update(obsoletion_candidate_child_of_branch)
+
+            # Column F: Other direct parents in Branch = other_parents_in_branch
+            other_parents_in_branch = parents_of_child & branch_descendants_set
+
+            # Column G: Other direct parents NOT in Branch = other_parents_not_in_branch
+            other_parents_not_in_branch = parents_of_child - other_parents_in_branch
+
+            all_ancestors = (
+                set(
+                    OI.ancestors(
+                        [child], predicates=[IS_A, PART_OF], method="ENTAILMENT"
+                    )
+                )
+                # - parents_of_child
+                - {child}
```

Comment:
What does the `{` syntax mean here? Just educational question for me

Reply:
set. `child` at this point is just a CURIE. In order for the subtraction to work, I have to put it in a set.

Reply:
Oops didn't know that `{` was the set operator! Thanks!